src/components/Books/InputBook.js

```diff
@@ -1,17 +1,41 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../../redux/books/bookSlice';
 import './books.css';
 
-const InputBook = () => (
-  <div className="input__container">
-    <h1>ADD NEW BOOK</h1>
-    <form>
-      <input type="text" name="name" placeholder="Book title..." required />
-      <input type="text" name="name" placeholder="Author..." required />
-      <button className="add__btn" type="submit">
-        ADD BOOK
-      </button>
-    </form>
-  </div>
-);
+const InputBook = () => {
+  const [state, setState] = useState({
+    title: '',
+    author: '',
+  });
+
+  const dispatch = useDispatch();
+
+  const handleChange = (e) => {
+    setState({
+      ...state,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleSubmit = () => {
+    dispatch(addBook(state.title, state.author));
+    state.title = '';
+    state.author = '';
```

Comment:
- [ ] **[OPTIONAL]:** I highly suggest that you please consider modifying the state using the `setState` function instead of assigning a value directly! as the `setState` will schedule an update to your state and will re-render your component, following React's conventions! 🤓 🔧

Reply:
@anagudelogu Thank you for the insight☺